src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -10,6 +10,20 @@
 
 namespace ProtoFFI
 {
+
+    internal class TypeEqualityComparer : IEqualityComparer<Type>
+    {
+        public bool Equals(Type x, Type y)
+        {
+            return x.IsEquivalentTo(y) || x.Equals(y);
+        }
+
+        public int GetHashCode(Type obj)
+        {
+            return obj.GetHashCode();
```

Comment:
The hash code will be different for 2 equivalent embedded COM types.
So this strategy does not work

Reply:
I could implement a custom dictionary that does not use hash codes. but the current dictionaries are already exposed as public. It would be complicated